src/transaction/log_page_buffer.c

```diff
@@ -6946,6 +6945,19 @@ logpb_remove_archive_logs_exceed_limit (THREAD_ENTRY * thread_p, int max_count)
       if (last_arv_num_to_delete >= first_arv_num_to_delete)
 	{
 	  log_Gl.hdr.last_deleted_arv_num = last_arv_num_to_delete;
+
+	  /* Update the last_blockid needed for vacuum. */
+	  new_page_id = logpb_Arv_page_info_table.page_info[last_arv_num_to_delete + 1].start_pageid;
```

Comment:
I don't think it is correct. Please notice that `last_arv_num_to_delete` could not be used as index of the page table.

Reply:
please also notice that you may not find the archive volume from the page table, where overflow of entries of table happens.